Add Log Delete Command and Parser

sdevih opened the pull request:
The logdelete command class and parser class are added to implement the delete functionality of logs.

The addressbook is modified to parse the log delete command in the CLI.

Closes #82

sdevih committed:
Add Log Delete Command and Parser

The logdelete command class and parser class are added to implement the
delete functionality of logs.

The addressbook is modified to parse the lgo delete command in the CLI.

sdevih committed:
Add test cases for LogDeleteCommand

Test cases are added to check if the logDeleteCommand is functioning correctly.

sdevih committed:
Update timeServed when log is deleted

The LogDeleteCommand was fixed to ensure the timeServedvalue in
volunteer and befriendee is updated with the correct value
when a log is deleted.

The test cases for LogDeleteCommand will be updated after.

sdevih committed:
Fix LogDelete and add tests

The LogDeleteCommand is updated to work with the latest log feature.

The test cases for the LogDeleteCommand are updated.

sdevih committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2103T-T09-3/tp into branch-deleteLogCommand

sdevih committed:
Add test cases for LogDelete and update UG

The test cases for LogDeleteCommand class was fixed to accommodate the
latestLog feature and timeServed feature.

The test cases for LogDeleteCommandParser and AddressBook parser was added.

The User Guide was updated with information on the log delete
functionality.

chaaaaun commented on src/main/java/scrolls/elder/logic/commands/LogDeleteCommand.java:
```diff
@@ -0,0 +1,152 @@
+
+
+        Person updatedVolunteer = createUpdatedPerson(volunteer, durationToDelete, latestLogIdVolunteer);
+        Person updatedBefriendee = createUpdatedPerson(befriendee, durationToDelete, latestLogIdVolunteer);
```
Probably meant to use `latestLogIdBefriendee`?

chaaaaun commented on src/main/java/scrolls/elder/logic/commands/LogDeleteCommand.java:
```diff
@@ -0,0 +1,152 @@
+        Person befriendee = lastShownPList.get(logToDelete.getBefriendeeId());
+
+        Integer latestLogIdBefriendee = getLatestLogId(befriendee, logToDelete, logStore, logToDelete.getLogId());
+        Integer latestLogIdVolunteer = getLatestLogId(volunteer, logToDelete, logStore, logToDelete.getLogId());
```
We should want to return the next latest log ID if the log being deleted happens to be the latest, currently it just returns the ID of the log being deleted.

chaaaaun reviewed (approved):
Might have to rework the `getLatestLogId` part a little but otherwise looks good.

sdevih committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2103T-T09-3/tp into branch-deleteLogCommand

sdevih committed:
Fix log delete command to display recent log

The log delete command was fixed to ensure that the next recent log is
displayed if the deleted log was displayed as the latest log.

The test cases for the log delete class was fixed to accommodate this
change.

sdevih committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2103T-T09-3/tp into branch-deleteLogCommand

chaaaaun reviewed (approved):
Thanks for the changes, LGTM!
